prometheus/go_collector.go

```diff
@@ -236,6 +242,9 @@ func (c *goCollector) Describe(ch chan<- *Desc) {
 func (c *goCollector) Collect(ch chan<- Metric) {
 	c.goroutines.Set(float64(runtime.NumGoroutine()))
 	ch <- c.goroutines
+	n, _ := runtime.ThreadCreateProfile(nil)
```

Comment:
Are you sure this reliably produces the number of OS threads? From what I can see, it could also return the number of OS threads _created_ but wouldn't be adjusted if an OS thread gets _destroyed_. But I don't know any details. That's why I'm asking.